### DISCUSSION TITLE
Makefile for Windows

### DISCUSSION THREAD

**Discussion by cbruccoleri** [mbitsnbites/vasm-mirror#21] [General]
Hello,
I am trying to compile VASM x86 architecture on Windows 10. I have the Windows SDK compiler (cl.exe) and nmake installed.

When I run the following command at the command prompt (native x86 command prompt for VS 2019)

`nmake /F Makefile.Win32 CPU=6502 SYNTAX=std`

I get the following error:

`C:\local\code\vasm\vasm.c : fatal error C1083: Cannot open compiler generated file: 'C:\local\code\vasm\obj_win32\6502_std_vasm.o': No such file or directory`

This can be fixed by manually creating the folder `obj_win32`; the Makefile should be changed to create such folder automatically.

Trying compilation again, I get the following error:
`C:\Program Files (x86)\Windows Kits\10\include\10.0.18362.0\um\winnt.h(2487): error C2118: negative subscript
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.22.27905\bin\HostX86\x86\cl.EXE"' : return code '0x2'
`
This is due to an ASSERT in the `winnt.h` header file. This problem can be fixed by adding the following symbol in the Makefile.Win32 (line 9):

-DWINDOWS_IGNORE_PACKING_MISMATCH

I am not sure whether this may cause instability in the compiled executable; perhaps a more detailed review should be done by the code maintainer, but with these changes, the code compiles on Windows 10. Thank you for VASM!

**Comment by mbitsnbites** [comment]
This is a mirror. Please contact upstream: http://sun.hasenbraten.de/vasm/
